packages/block-library/src/social-link/edit.native.js

```diff
@@ -65,8 +65,8 @@ const SocialLinkEdit = ( {
 
 	const animatedValue = useRef( new Animated.Value( 0 ) ).current;
 
-	const IconComponent = getIconBySite( service )();
-	const socialLinkName = getNameBySite( service );
+	const IconComponent = GetIconBySite( service )();
```

Comment:
React hooks like `useSelect` won't work as expected this way, they need to be rendered as components.

Reply:
Actually, those could be React hooks:

```js
const socialLinkIcon = useServiceIcon( service );
const socialLinkName = useServiceName( service );
```

Reply:
I fixed this in 0edca09 by removing the `()` and using the `IconComponent` directly in render

Reply:
React docs discourage nesting hooks in regular functions:

https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

> **Don’t call Hooks inside loops, conditions, or nested functions.** Instead, always use Hooks at the top level of your React function, before any early returns. By following this rule, you ensure that Hooks are called in the same order each time a component renders.

I'm not sure why they advocate for that but it as simple as replacing `Get` with `use` in function names.

Reply:
> I'm not sure why they advocate for that

Abramov's _[Why do hooks rely on call order?](https://overreacted.io/why-do-hooks-rely-on-call-order/)_ is a great starting point. Some of his other posts ([1](https://overreacted.io/algebraic-effects-for-the-rest-of-us/), [2](https://overreacted.io/why-isnt-x-a-hook/)) provide deep complementary insight.

Reply:
@gziolo I switched the two functions to hooks :+1: